### PR TITLE
Add Winograd Conv method for CPU

### PR DESCRIPTION
**Pull request by wuhuikx**
Add winograd conv method. Users can select the direct conv or winograd conv in the model file.

### CODE REVIEW

**Comment by ezyang** [comment]
CC @yinghai

**Comment by wuhuikx** [comment]
@yinghai @jgong5

**Review by yinghai — CHANGES_REQUESTED** [review]

File: caffe2/ideep/operators/conv_pool_base_op.h [hunk 1]
```diff
@@ -5,6 +5,9 @@
 
 namespace caffe2 {
 
+#define CONV_ALGORITHM_AUTO 0
```
[the remark this answers is not in the document]
Comment (wuhuikx) [reply caffe2/ideep/operators/conv_pool_base_op.h]: Have changed the macros to constexpr and move them to  caffe2/ideep/ideep_utils.h file.
The patch #9488 will also use the constexpr.

**Review by yinghai — COMMENTED** [review]
Btw, do we have tests for this?

**Comment by wuhuikx** [comment]
Yes, we have test this for several models.

**Comment by yinghai** [comment]
I mean localized unit tests?

**Comment by yinghai** [comment]
Could you check the test failures? Rebase maybe?

**Comment by wuhuikx** [comment]
> Could you check the test failures? Rebase maybe?

We depend on this PR https://github.com/pytorch/pytorch/pull/12164

**Pull request closed without merging**